gui/gui/src/TGFileBrowser.cxx

```diff
@@ -1225,10 +1228,21 @@ void TGFileBrowser::DoubleClicked(TGListTreeItem *item, Int_t /*btn*/)
    if (obj && !obj->InheritsFrom("TSystemFile")) {
       TString ext = obj->GetName();
       if (obj->InheritsFrom("TDirectory") && (obj->IsA() != TClass::Class())) {
-         if (((TDirectory *)obj)->GetListOfKeys())
-            fNKeys = ((TDirectory *)obj)->GetListOfKeys()->GetEntries();
-         else
-            fNKeys = 0;
+         if (obj->TestBit(9/*TDirectoryFile::kCustomBrowse*/)) {
+            is_rntuple = kTRUE;
+            static void *rbrowser = 0;
+            if (!rbrowser) rbrowser = (void *)gROOT->ProcessLine(TString::Format("new ROOT::Experimental::RNTupleBrowser((TDirectory *)%#tx);", (uintptr_t)obj));
```

Comment:
If the class and its destructor are available (see line 264) why not use its constructor directly? (or conversely should the delete be done via gROOT->ProcessLine)?

Reply:
And also consider forward declaring RNTupleBrowser and avoid the 'void*'.

Reply:
Now delete is also done with gROOT->ProcessLine. TGFileBrowser.cxx shouldn't be able to call functions defined in RNTupleBrowser.cxx, but it could do so only for the destructor. (maybe there were some compiler optimisations or so?) So to not rely on something like that ProcessLine is used now.
Originally I forward declared the class, but I found it ugly writing `namespace ROOT{ namespace Experimental{ class RNTupleBrowser } //end Experimental } // end ROOT` just for this one class, so I used a void pointer instead. But now it's reverted back to the class forward declaration. (and also got rid of the redundant `static void *rbrowser = 0`)